### ISSUE TITLE
Improve performance of skipped log levels

### ISSUE THREAD

**Issue by muxspace** [zatonovo/futile.logger#55]
If a log message is below current threshold, exit early and avoid evaluation to improve performance.

**Comment by mvanwyk** [comment]
@muxspace If you can wait until next week I can pick up this task for you

**Comment by iwccsbfb** [comment]
@muxspace @mvanwyk From this function (https://github.com/zatonovo/futile.logger/blob/master/R/logger.R#L133) , we are already doing early exit if a log message is below current threshold, (unless we set carp=TRUE, but this is expected), aren't we?

**Comment by muxspace** [comment]
@iwccsbfb The actual problem is with `flog.namespace`. See below for performance numbers.

Setup.
```
library(lambda.tools)
library(futile.logger)
m <- matrix(rnorm(10000), nrow=100)
```
Default threshold is `INFO`. Printing this message 10000 times takes almost 15 seconds.
```
> system.time(fold(1:10000, function(i,a) { flog.info("[%s] sum diag: %s", i, sum(diag(m)));a }, NULL))
   user  system elapsed 
 14.616   0.160  14.783
```
Calling `flog.debug` doesn't print any messages but still takes 3 seconds, so something is eating up resources.
```
> system.time(fold(1:10000, function(i,a) { flog.debug("[%s] sum diag: %s", i, sum(diag(m)));a }, NULL))
   user  system elapsed 
  3.364   0.000   3.367
```
Using `sprintf` results in order of magnitude better performance.
```
> system.time(fold(1:10000, function(i,a) { sprintf("[%s] sum diag: %s", i, sum(diag(m)));a }, NULL))
   user  system elapsed 
  0.112   0.000   0.114
```
Here is `flog.namespace` on its own. Note how this is pretty close to the `flog.debug` call at `INFO` threshold.
```
> system.time(fold(1:10000, function(i,a) { flog.namespace(); a }, NULL))   user  system elapsed 
  3.052   0.000   3.052 
```

**Comment by iwccsbfb** [comment]
Fixed in #60,  performance before the fix:
```
> system.time(fold(1:10000, function(i,a) { flog.namespace(); a }, NULL)) 
   user  system elapsed 
  4.812   0.008   4.847 
```

After the fix:
```
> system.time(fold(1:10000, function(i,a) { flog.namespace(); a }, NULL)) 
   user  system elapsed 
  0.880   0.008   0.908 
```
